src/templates/csharp/TestBase.cs

```diff
@@ -0,0 +1,1066 @@
+
+            SwipeByPoint(fromPoint, toPoint, durationInMs);
+        }
+
+        /**
+         * Swipe from Point to Point (with accelerate)
+         */
+        public void SwipeByPoint(Point fromPoint, Point toPoint, int durationInMs)
+        {
+            Console.WriteLine($"Swipe from point ({fromPoint.X}, {fromPoint.Y}) to point ({toPoint.X}, {toPoint.Y}) with duration {durationInMs}");
+
+            TouchAction swipe = new TouchAction(driver);
+            swipe.Press(fromPoint.X, fromPoint.Y).Wait(durationInMs).MoveTo(toPoint.X, toPoint.Y).Release();
+
+            MultiAction multiAction = new MultiAction(driver);
+            multiAction.Add(swipe).Perform();
+        }
+
+        /**
+         * Quick swipe to top from a Point
+         */
+        public void SwipeToTop(Point fromPoint)
+        {
+            Point toPoint = new Point(fromPoint.X, GetScreenSize().Y - 10);
+            Console.WriteLine($"Swipe to top from point ({fromPoint.X}, {fromPoint.Y}) to point ({toPoint.X}, {toPoint.Y})");
+
+            SwipeByPoint(fromPoint, toPoint, 100);
+        }
+
+        /**
+         * Drag from Point to Point (no accelerate)
+         */
+        public void DragByPoint(Point fromPoint, Point toPoint)
+        {
+            Console.WriteLine($"Drag from point ({fromPoint.X}, {fromPoint.Y}) to point ({toPoint.X}, {toPoint.Y})");
+
+            TouchAction swipe = new TouchAction(driver);
+            swipe.Press(fromPoint.X, fromPoint.Y).Wait(300).MoveTo(toPoint.X, toPoint.Y).Release();
+
+            MultiAction multiAction = new MultiAction(driver);
+            multiAction.Add(swipe).Perform();
+        }
+
+        public void SendKeys(string keys)
+        {
+            sleep(Config.SleepTimeBeforeSendKeysInMs);
+
+            Console.WriteLine($"Send keys: {keys}");
+
+            if (this.isIos) {
+                var uri = new UriBuilder($"{GetAppiumServerUrl()}/session/{driver.SessionId}/keys");
+                var query = new Dictionary<string, string>
+                {
+                    {"value", keys}
+                };
+                uri.Query = new FormUrlEncodedContent(query).ReadAsStringAsync().Result;
+                var request = new HttpRequestMessage{
+                    Method = HttpMethod.Post,
+                    RequestUri = uri.Uri,
+                    Headers = {
+                        {
+                            HttpRequestHeader.Authorization.ToString(), Config.GetBasicAuthString()
+                        }
+                    }
+                };
+                httpClient.SendAsync(request).Wait();
+            } else
+            { 
+                for (int index = 0; index < keys.Count(); index++)
+                {
+                    int codePoint = Char.ConvertToUtf32(keys, index);
+                    GetAndroidDriver().PressKeyCode(codePoint);
+                    sleep(SleepAfterAction);
+                }
+            }
```

Comment:
SendKeys is not work as expected on Android. It send unexpected key code that make the application go to background.

Java implementation:
<img width="553" alt="image" src="https://github.com/kobiton/appium-script-generator/assets/15189024/5678f1f7-521d-44cb-8107-6e32ff075db1">